spyder/plugins/editor/widgets/editorstack/editorstack.py

```diff
@@ -653,7 +655,11 @@ def update_fname_label(self):
         """Update file name label."""
         filename = to_text_string(self.get_current_filename())
         if len(filename) > 100:
-            shorten_filename = u'...' + filename[-100:]
+            metrics = QFontMetrics(self.default_font)
```

Comment:
```suggestion
            metrics = QFontMetrics(
                self.default_font
                if self.default_font is not None
                else self.font()
            )
```

This will fix the errors in our tests because when running them `default_font` can not be set.